subprojects/test-runner/device-provider/impl/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/KubernetesReservationState.kt

```diff
@@ -20,42 +25,67 @@ internal class KubernetesReservationState(
 ) {
 
     private val requests = PriorityBlockingQueue<PodRequest>()
+    private var state = INITIAL
+    private val lock = Mutex()
 
-    fun claim(requestedPodCount: Int) {
-        val claimTime = timeProvider.nowInstant()
-        requests.addAll(
-            (0 until requestedPodCount).map { PodRequest(claimTime) }
-        )
+    suspend fun claim(requestedPodCount: Int) {
```

Comment:
```suggestion
    suspend fun claim(requestedPodCount: Int) = lock.withLock {
```
For the sake of consistency with other methods.